homeassistant/components/energyid/config_flow.py

```diff
@@ -0,0 +1,365 @@
+from .const import (
+    CONF_DEVICE_ID,
+    CONF_DEVICE_NAME,
+    CONF_PROVISIONING_KEY,
+    CONF_PROVISIONING_SECRET,
+    DOMAIN,
+)
+from .subentry_flow import EnergyIDSubentryFlowHandler
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_ENERGYID_DEVICE_NAME_FOR_WEBHOOK = "Home Assistant"
+ENERGYID_DEVICE_ID_FOR_WEBHOOK_PREFIX = "homeassistant_eid_"
+
+
+def _generate_energyid_device_id_for_webhook() -> str:
+    """Generate a unique device ID for this Home Assistant instance to use with EnergyID webhook."""
+    return f"{ENERGYID_DEVICE_ID_FOR_WEBHOOK_PREFIX}{secrets.token_hex(4)}"
+
+
+class EnergyIDConfigFlow(ConfigFlow, domain=DOMAIN):
+    """Handle the configuration flow for the EnergyID integration."""
+
+    VERSION = 1
+    _config_entry_being_reconfigured: ConfigEntry | None = None
+
+    def __init__(self) -> None:
+        """Initialize the config flow with default flow data."""
+        self._flow_data: dict[str, Any] = {
+            "provisioning_key": None,
+            "provisioning_secret": None,
+            "webhook_device_id": _generate_energyid_device_id_for_webhook(),
+            "webhook_device_name": DEFAULT_ENERGYID_DEVICE_NAME_FOR_WEBHOOK,
+            "claim_info": None,
+            "record_number": None,
+            "record_name": None,
+        }
+
+    async def _perform_auth_and_get_details(self) -> str | None:
+        """Authenticate with EnergyID and retrieve device details."""
+        if (
+            not self._flow_data["provisioning_key"]
+            or not self._flow_data["provisioning_secret"]
+        ):
+            _LOGGER.error("Missing credentials for authentication")
+            return "missing_credentials"
+
+        _LOGGER.debug(
+            "Attempting authentication with device ID: %s, device name: %s",
+            self._flow_data["webhook_device_id"],
+            self._flow_data["webhook_device_name"],
+        )
+
+        session = async_get_clientsession(self.hass)
+        client = WebhookClient(
+            provisioning_key=self._flow_data["provisioning_key"],
+            provisioning_secret=self._flow_data["provisioning_secret"],
+            device_id=self._flow_data["webhook_device_id"],
+            device_name=self._flow_data["webhook_device_name"],
+            session=session,
+        )
+
+        try:
+            is_claimed = await client.authenticate()
+        except ClientError:
+            _LOGGER.warning(
+                "Connection error during EnergyID authentication", exc_info=True
+            )
+            return "cannot_connect"
+        except RuntimeError:
+            _LOGGER.exception("Unexpected runtime error during EnergyID authentication")
+            return "unknown_auth_error"
+
+        if is_claimed:
+            self._flow_data["record_number"] = client.recordNumber
+            self._flow_data["record_name"] = client.recordName
+            self._flow_data["claim_info"] = None
+            _LOGGER.info(
```

Comment:
```suggestion
            _LOGGER.debug(
```
info is reserved for the system